boto3-assume/aws/GdsAwsClient.py

```diff
@@ -0,0 +1,194 @@
+# GdsAwsClient
+# Manage sts assume-role calls and temporary credentials
+import os
+import boto3
+import json
+import configparser
+
+class GdsAwsClient:
+
+    # initialise empty dictionaries for clients and assume role sessions
+    resources = dict()
+    clients = dict()
+    sessions = dict()
+
+    # retrieve default key and secret from ~/.aws/credentials file
+    def load_credentials(self): 
```

Comment:
we should try to read these from env variables, which are injected by `aws-vault`